### PR TITLE
Fix assembly binding redirection in AnalyzerRunner

### PR DESCRIPTION
**Pull request by sharwell**
Fixes the signed build by ensuring the correct assembly binding redirection is in place for AnalyzerRunner.

### CODE REVIEW

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
 <Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\..\build\Targets\GenerateCompilerExecutableBindingRedirects.targets" />
```
Comment (heejaechang) [inline src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: just curious. how this binding redirect works?

**Comment by heejaechang** [comment]
@sharwell do we have a wiki or something on how to use this tool? would like to take a look and see some perf trace.

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj (hunk 1 above)
Comment (sharwell) [reply src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: @jasonmalinowski ⬆️

**Comment by sharwell** [comment]
> do we have a wiki or something on how to use this tool

#23582
#23583
#26778

The tool also has several predefined launch profiles:

https://github.com/dotnet/roslyn/blob/2acdc8668ee75c3380cad237a126087bdab8dae3/src/Tools/AnalyzerRunner/Properties/launchSettings.json#L3-L30

**Comment by heejaechang** [comment]
awesome! thank you

File: src/Tools/AnalyzerRunner/AnalyzerRunner.csproj (hunk 1 above)
Comment (jasonmalinowski) [reply src/Tools/AnalyzerRunner/AnalyzerRunner.csproj]: There are core MSBuild targets that will automatically generate binding redirects; the usual process is ResolveAssemblyReferences looks across all your dependencies, figures out the versions, and updates your app.config during build. You can also pass some "hints" in which say "assume I have this version or that version around I need to redirect from". That file just contains those hints.

**Pull request merged**